dashboard/app/views/projects/index.html.haml

```diff
@@ -69,6 +69,10 @@
       $scope.projectsLoaded = true;
     }).catch($scope.genericError);
 
+    $scope.projectVisible = function(project) {
+      return (!project.hidden);
```

Comment:
Yeah, adding filter/search to the API may make sense, but I think if we do
it we will want it to be generic filter/search because from the API's point
of view the properties are not meaningful (so basically the client would
give the server a query), and I can see us wanting to filter on other
things (like projectTemplate name), not just hidden. We could also add
filtering to the js "service" (ie. the projectsService js object) first if
we think it's messy to do this in the view.

On Fri, Feb 20, 2015 at 1:39 PM, Josh Lory <notifications@github.com> wrote:

> In dashboard/app/views/projects/index.html.haml
> <https://github.com/code-dot-org/code-dot-org/pull/1360#discussion_r25105301>
> :
>
> > @@ -69,6 +69,10 @@
> >        $scope.projectsLoaded = true;
> >      }).catch($scope.genericError);
> >
> > +    $scope.projectVisible = function(project) {
> > +      return (!project.hidden);
>
> Either a showHidden=true param or maybe we really do need namespaces that
> can be used as filters in the .all() query. This feels like scope creep so
> I'm fine with client-side for now, until we better understand what the API
> filters look like.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/1360/files#r25105301>.
>